Mobile-Development-and-React-Native.md

```diff
@@ -91,8 +91,54 @@ The values that can be used are:
 -   `sentences`: first letter of each sentence (default)
 -   `none`: do not capitalize automatically
 
+## Optimization
+
+`Before run into the optimization rabbit role in your applications:`
+> Measure. Don’t tune for speed until you’ve measured, and even then don’t unless one part of the code overwhelms the rest. - [Rob Pyke, 5 Rules of Programming](http://users.ece.utexas.edu/~adnan/pike.html)
+
+### Images
+
+Allways try to use the smallest possible image size. The larger the file, the longer they will take to transfer and display on the screen. Use image resizing software to compress the image. Remember to find a good balance between the quality and size of the photo.
+
+`Note: some image pickers has the built-in option to apply some compress or resize to the image.`
+
+#### Image size and suffixes
+
+You can also use the suffix @2x and @3x, which will be displayed depending on the screen resolution of your device.
+
+Example:
+```
+|src
+|->assets
+|  -> logo.png
+|  -> logo@2x.png
+|  -> logo@3x.png
+|->App.js
+```
+
+```jsx
+// App.js
+
+import logo from '~/assets/logo.png';
+
+export default function App() {
+
+  return (
+    <Image source={logo}>
+  );
+}
+```
+
+In the above example if your app are runnning your app on an `iPhone 7` the React Native will use the `logo@2x.png` file, while on an `iPhone 7 Plus` the `logo@3x.png` will be used. If the image size do note match the device screen, the best option will be selected.
+
+#### Loading Static Assets
+
+When loading asssets from the JavaScript bundle, React Native first imports the file and then sends it to Native UI (which is responsible for drawing the interface). If your assets are located in the native modules, React Native will simply tell the Native UI where the asset is located (`You can see the difference in the React Native architecture image below`).
 
+![React Native architecture](./assets/mobile-development/rn-architecture.jpeg)
 
+So, depending on where your image is being used, and how often it is required by your application (like an icon), place it inside the native module would improve the UI loading.
 
+To see how to use both aproches you can take a look at the [official documentation](https://reactnative.dev/docs/images#static-image-resources).
```

Comment:
A couple more:
- `(...) your application (like an icon), *placing* it inside the native module would (...)`
- `To see how to use both *approaches* you can take (...)`